### PR TITLE
Moved Google Dataflow to be externally configured

### PR DESCRIPTION
**Pull request by herohde**
* Added the ability to set subservice name, if autodetection of
   subservice is disabled.
 * Added the ability to configure static labels.

### CODE REVIEW

File: lib/fluent/plugin/out_google_cloud.rb [hunk 1]
```diff
@@ -97,6 +98,11 @@ class GoogleCloudOutput < BufferedOutput
     #   }
     config_param :label_map, :hash, :default => nil
 
+    # labels (specified as a JSON object) is a set of custom labels
+    # provided at configuration time. It allows users to inject extra
+    # environmental information into every message.
```
Comment (mr-salty) [inline lib/fluent/plugin/out_google_cloud.rb]: mention that it is a map of label name -> label value and give an example (similar to label_map above).

File: lib/fluent/plugin/out_google_cloud.rb [hunk 2]
```diff
@@ -254,6 +260,8 @@ def configure(conf)
               cluster_name_from_kube_env(@kube_env)
             detect_cloudfunctions(attributes)
           end
+        elsif @subservice_name
```
Comment (mr-salty) [inline lib/fluent/plugin/out_google_cloud.rb]: I think this should probably override 'detect_subservice' to make it more consistent with the way the other parameters work (i.e. 'project_id' is always used regardless of 'use_metadata_service').

Document this where the parameter is declared.

File: lib/fluent/plugin/out_google_cloud.rb (hunk 1 above)
Comment (herohde) [inline lib/fluent/plugin/out_google_cloud.rb]: Done

File: lib/fluent/plugin/out_google_cloud.rb (hunk 2 above)
Comment (herohde) [inline lib/fluent/plugin/out_google_cloud.rb]: Done

**Comment by herohde** [comment]
Thanks Todd. PTAL

File: lib/fluent/plugin/out_google_cloud.rb [hunk 3]
```diff
@@ -63,13 +62,16 @@ class GoogleCloudOutput < BufferedOutput
...
-    # Whether to try to detect of the VM is owned by a "subservice" such as App
+    # Whether to try to detect if the VM is owned by a "subservice" such as App
     # Engine of Kubernetes, rather than just associating the logs with the
     # compute service of the platform. This currently only has any effect when
     # running on GCE.
+    #
     # The initial motivation for this is to separate out Kubernetes node
     # component (Docker, Kubelet, etc.) logs from container logs.
     config_param :detect_subservice, :bool, :default => true
+    # The subservice_name overrides the subserice detection, if provided.
```
Comment (mr-salty) [inline lib/fluent/plugin/out_google_cloud.rb]: s/subserice/subservice/

File: lib/fluent/plugin/out_google_cloud.rb [hunk 4]
```diff
@@ -147,6 +163,11 @@ def configure(conf)
 
       # TODO: Send instance tags as labels as well?
       @common_labels = {}
+      if @labels
+        @labels.each do |key, value|
```
Comment (mr-salty) [inline lib/fluent/plugin/out_google_cloud.rb]: sorry I missed this before but I think you can just write this as ```@common_labels.merge!(@labels)```

Comment (mr-salty) [inline lib/fluent/plugin/out_google_cloud.rb]: (in which case rubocop will probably tell you to write it all on one line as ```@common_labels.merge!(@labels) if @labels```)

**Comment by mr-salty** [comment]
LG otherwise - thanks!

File: lib/fluent/plugin/out_google_cloud.rb (hunk 3 above)
Comment (herohde) [inline lib/fluent/plugin/out_google_cloud.rb]: Done

File: lib/fluent/plugin/out_google_cloud.rb (hunk 4 above)
Comment (herohde) [inline lib/fluent/plugin/out_google_cloud.rb]: Done

**Comment by herohde** [comment]
Thanks!

**Comment by mr-salty** [comment]
LGTM

**Pull request merged**